lib/db-utils.js

```diff
@@ -207,12 +214,19 @@ module.exports = (srf, logger) => {
           const [a] = await pp.query(sqlAccountBySid, [[matches[0].account_sid]]);
           if (0 === a.length) return failure;
           logger.debug({matches}, `found registration carrier using ${uri.host} and ${uri.user}`);
+          const sql =
+            `SELECT * FROM phone_numbers WHERE number = '${did}'
+            AND voip_carrier_sid = '${matches[0].voip_carrier_sid}'
+            AND account_sid = '${matches[0].account_sid}'`;
+          logger.debug({matches: matches[0], sql, did}, 'looking up DID');
+
+          const [r] = await pp.query(sql);
           return {
             fromCarrier: true,
             gateway: matches[0],
             service_provider_sid: a[0].service_provider_sid,
             account_sid: a[0].account_sid,
-            application_sid: matches[0].application_sid,
+            application_sid: r.length ? r[0].application_sid : matches[0].application_sid,
```

Comment:
Is it possible the query returned a row, but the row had a null value for application_sid?  Should the assignment instead be
```
application_sid: r[0]?.application_sid || matches[0].application_sid
```